content/blog/2021/09/towards-a-universal-policy-platform.md

```diff
@@ -0,0 +1,155 @@
+the lives of Operation teams. Simplicity is paramount when operating Kubernetes,
+having different ways to distribute and enforce policies goes into the opposite
+direction.
+
+The Kubewarden team wants to further expand the freedom of policy author, without
+having to compromise on the operation experience we currently provide.
+
+We want to provide a cohesive way to distribute policies, a uniform
+way to enforce them and a single platform to host them. All of that regardless of
+the language used to write these individual policies.
+
+We want to provide a **Universal Policy Platform**.
+
+## WebAssembly to the rescue
+
+Rego is a query language created by the Open Policy Agent project. The language
+is inspired by [Datalog](https://en.wikipedia.org/wiki/Datalog), its
+main purpose is to perform queries on some JSON input data and provide some
+output as response.
+
+The good news is that Rego programs can be built into WebAssembly modules using
+the official `opa` command line utility. This is documented in depth inside
+of the [official documentation](https://www.openpolicyagent.org/docs/latest/wasm/)
+of Open Policy Agent.
+
+While building a Rego program into WebAssembly is simple, running the resulting WebAssembly
+module requires more efforts. The good news is... we did the hard work!
+
+We wrote a Rust library that can be used to invoke Rego policies compiled into
+WebAssembly modules.
+This library is called Burrego (a burrito with a Rego filling... obvious isn't it?! 🌯 🤓)
+and can be found [here](https://github.com/kubewarden/policy-evaluator/tree/main/crates/burrego).
+
+The Rego language provides some [built-in functions](https://www.openpolicyagent.org/docs/latest/policy-reference/#built-in-functions)
+to perform String, Encoding, Regex, ... operations.
+When building a Rego policy into a WebAssembly module, some of these built-in
+functions are going to be implemented inside of the Wasm file itself; while others
+have to be provided at execution time by the WebAssembly runtime evaluating the module.
+
+The Burrego library implements the built-in functions that are not supported natively
+by the WebAssembly modules produced by `opa build`. There are still some built-ins that
+are not yet provided by Burrego, however, based on the policies we have seen in the open,
+the ones we already support should be enough for the majority of Kubernetes users.
+
+[This GitHub issue](https://github.com/kubewarden/policy-evaluator/issues/56)
+keeps track of the Rego built-ins we have still to implement. Feel free to
+comment over there to prioritize our work.
+
+## One language, two frameworks
+
+Open Policy Agent integrates with Kubernetes using the [kube-mgmt](https://github.com/open-policy-agent/kube-mgmt)
+component.
+More recently, a new Kubernetes integration gained popularity: [Gatekeeper](https://github.com/open-policy-agent/gatekeeper).
+
+Both kube-mgmt and Gatekeeper use Rego to write their policies. However
```

Comment:
kube-mgmt is just a sidecar that pulls other resources from the cluster and pushes them to the opa server, and that is able to read policies from ConfigMaps. In an OPA deployment policies are still served by `opa serve`, so I would not say "kube-mgmt use Rego to write its policies". Policies with OPA are written in Rego, served by`opa serve`, and the `kube-mgmt` sidecar provides some Kubernetes integration bits that pushes to the `opa server`.

Reply:
I get it, thanks. I've clarified